make install-dev-tools failure

## Summary

`make install-dev-tools` fails due to an incompatibility between `rustc` and `cargo-binstall` versions

## Steps to reproduce

1. Run `make install-dev-tools`

## Expected Result

Install dependencies

## Actual Result

```
make install-dev-tools
cargo install --locked dprint
    Updating crates.io index
     Ignored package `dprint v0.47.2` is already installed, use --force to override
cargo install cargo-llvm-cov
    Updating crates.io index
     Ignored package `cargo-llvm-cov v0.6.11` is already installed, use --force to override
cargo install cargo-hack
    Updating crates.io index
     Ignored package `cargo-hack v0.6.30` is already installed, use --force to override
cargo install cargo-udeps
    Updating crates.io index
     Ignored package `cargo-udeps v0.1.49` is already installed, use --force to override
cargo install flaky-finder
    Updating crates.io index
     Ignored package `flaky-finder v0.2.22` is already installed, use --force to override
cargo install cargo-nextest --locked
    Updating crates.io index
     Ignored package `cargo-nextest v0.9.72` is already installed, use --force to override
cargo install cargo-binstall
    Updating crates.io index
error: cannot install package `cargo-binstall 1.7.4`, it requires rustc 1.79.0 or newer, while the currently active rustc version is 1.78.0
`cargo-binstall 1.6.9` supports rustc 1.70.0
make: *** [install-dev-tools] Error 101
```

## Additional Information

We could update `rust-toolchain.toml` channel to `1.79.0` (implies updating CIs as well) or simply pin `cargo-binstall` to `1.6.9`

--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,7 @@ install-dev-tools:  ## Installs all necessary cargo helpers
 	cargo install cargo-udeps
 	cargo install flaky-finder
 	cargo install cargo-nextest --locked
-	cargo install cargo-binstall
+	cargo install --version 1.6.9 cargo-binstall
 	cargo binstall cargo-risczero
 ifeq ($(shell uname -ms), Darwin x86_64)
 	cargo risczero build-toolchain

